Sources/StytchCore/StytchClient/TOTP/StytchClient+TOTP.swift

```diff
@@ -0,0 +1,125 @@
+public extension StytchClient {
+    /// Time-based One-time Passcodes (TOTPs) are one-time passcodes that are generated based on a shared secret and the current time. TOTPs are also often referred to as Authenticator Apps and are a common form of secondary authentication. Creating a Stytch instance of a TOTP for a User creates a shared secret. This secret is shared by Stytch with the end user's authenticator application of choice (e.g. Google Authenticator). The authenticator app can then generate TOTPs that are valid for a specific amount of time (generally 30 seconds). The end user inputs the TOTP and the developer can use the authenticate method to verify that the TOTP is valid. To call these methods, TOTPs must be enabled in the [SDK Configuration page](https://stytch.com/dashboard/sdk-configuration) of the Stytch dashboard.
+    struct TOTP {
+        let router: NetworkingRouter<TOTPRoute>
+
+        /// Wraps Stytch's [create](https://stytch.com/docs/api/totp-create) endpoint. Call this method to create a new TOTP instance for a user. The user can use the authenticator application of their choice to scan the QR code or enter the secret.
+        public func create(parameters: CreateParameters) async throws -> CreateResponse {
+            try await router.post(to: .create, parameters: parameters)
+        }
+
+        /// Wraps Stytch's [authenticate](https://stytch.com/docs/api/totp-authenticate) endpoint. Call this method to authenticate a TOTP code entered by a user.
+        public func authenticate(parameters: AuthenticateParameters) async throws -> AuthenticateResponseType {
+            try await router.post(to: .authenticate, parameters: parameters) as AuthenticateResponse
+        }
+
+        /// Wraps Stytch's [recovery_codes](https://stytch.com/docs/api/totp-get-recovery-codes) endpoint. Call this method to retrieve the recovery codes for a TOTP instance tied to a user. Note: If a user has enrolled another MFA method, this method will require MFA. See the [Multi-factor authentication](https://stytch.com/docs/sdks/javascript-sdk#resources_multi-factor-authentication) section for more details.
+        public func recoveryCodes() async throws -> RecoveryCodesResponse {
+            try await router.post(to: .recoveryCodes, parameters: JSON())
```

Comment:
The API docs say we pass a userId, but this is an empty JSON object, I'm assuming there's some magic somewhere that adds that?

Reply:
Web-backend uses the session auth header to get the user id and pass that to the api ([here](https://github.com/stytchauth/web-backend/blob/main/app/sdk/totps.ts#L88)). The expected parameters are [here](https://github.com/stytchauth/web-backend/blob/main/app/sdk/totps.ts#L72)

Reply:
BTW, when you need it, I've laid out the little bit of TOTP info [here](https://docs.google.com/document/d/1OcCESxjfre5ztDm7qp57tbAOZyMyVfqbHlrbwbwXJWQ/edit#heading=h.eqdbtak2xz9l)